hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashPolicyOzone.java

```diff
@@ -98,7 +105,8 @@ public Runnable getEmptier() throws IOException {
 
     @Override
     public void run() {
-      if (emptierInterval == 0) {
+      // if this is not the leader node,don't run the emptier
+      if (emptierInterval == 0 || !om.isLeader()) {
```

Comment:
I think this should happen after the thread.sleep at line 117. As for most of the cases, when an OM restart the trash emptier it should check this status and skip doing any Trash processing.

Reply:
Also lets see if we can add a unit test for this change. Please check with @bharatviswa504 on the test